src/OrchardCore.Modules/OrchardCore.Contents/Indexing/ContentItemIndexCoordinator.cs

```diff
@@ -49,7 +51,9 @@ public async Task BuildIndexAsync(BuildIndexContext context)
                 var partActivator = _contentPartFactory.GetTypeActivator(partTypeName);
                 var part = (ContentPart)context.ContentItem.Get(partActivator.Type, partName);
 
-                var typePartIndexSettings = contentTypePartDefinition.GetSettings<ContentIndexSettings>();
+                MethodInfo ContentTypePartDefinitionMethod = typeof(ContentTypePartDefinition).GetMethod("GetSettings");
+                MethodInfo ContentTypePartDefinitionGeneric = ContentTypePartDefinitionMethod.MakeGenericMethod(context.Settings.GetType());
+                var typePartIndexSettings = (IContentIndexSettings)ContentTypePartDefinitionGeneric.Invoke(contentTypePartDefinition, null);
```

Comment:
Might not have to be an interface, the existing type might work as well.

Reply:
Note to myself, it can't be a common "ContentIndexSettings" property since you might want different values when two index providers are enabled.